### ISSUE TITLE
File Copy Now Fail On Envrioment.

### ISSUE THREAD

**Issue by russelldyas** [Microsoft/vso-agent-tasks#849]
Previous to the weekend I had File Copy working to azure machine in VSO release manager.  The way I had it setup was:

I had created a machine group in "test" section of VSO and assigned machines to that group.  Then I had put the group name under what was then group name.  However over weekend interface has had an update.  Now if try and use anything under machine be that IP or the group name it fails out below.

As said this was working before the weekend so must be related to new interface update etc.  I also tried putting IP address direct , 40.X.X. Etc. etc.  But still get the same problem?

Regards

Russell

Logs From Release Manager

2015-11-23T12:47:38.9405498Z Executing the powershell script: C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\default\tasks\WindowsMachineFileCopy\1.0.14\WindowsMachineFileCopy.ps1
 
2015-11-23T12:47:40.1961763Z ##[debug]Entering script WindowsMachineFileCopy.ps1
 
2015-11-23T12:47:40.1991829Z ##[debug]environmentName = QA
 
2015-11-23T12:47:40.2001766Z ##[debug]adminUserName = qa-client-x-03\Rencore
 
2015-11-23T12:47:40.2011759Z ##[debug]resourceFilteringMethod = machineNames
 
2015-11-23T12:47:40.2011759Z ##[debug]machineNames = 
 
2015-11-23T12:47:40.2021819Z ##[debug]sourcePath = C:\a\a162196ff\Master SPCAF.TeamBuild\drop\Setups
 
2015-11-23T12:47:40.2031751Z ##[debug]targetPath = c:\rencore_msi
 
2015-11-23T12:47:40.2038897Z ##[debug]additionalArguments = 
 
2015-11-23T12:47:40.2045353Z ##[debug]copyFilesInParallel = true
 
2015-11-23T12:47:40.2051869Z ##[debug]cleanTargetBeforeCopy = false
 
2015-11-23T12:48:20.8881721Z ##[debug]Starting Register-Environment cmdlet call for environment : QA with filter 
 
2015-11-23T12:48:20.9757629Z ##[error]A parameter cannot be found that matches parameter name 'EnvironmentName'.
 
2015-11-23T12:48:20.9767829Z ##[debug]Completed Register-Environment cmdlet call for environment : QA
 
2015-11-23T12:48:20.9807873Z ##[debug]Starting Get-EnvironmentResources cmdlet call on environment name: 
 
2015-11-23T12:48:20.9848043Z ##[error]Cannot bind argument to parameter 'EnvironmentName' because it is null.
 
2015-11-23T12:48:20.9857811Z ##[debug]Completed Get-EnvironmentResources cmdlet call for environment name: 
 
2015-11-23T12:48:21.1579679Z ##[error]No machine exists under environment: 'QA' for deployment

**Comment by lrt512** [comment]
Having the same problem with PowerShell on Target Machines. It was working on Thursday, broken today. I created a build that is just a remote invocation, running a NOP tets.ps1... same problem.

2015-11-23T14:43:01.0004041Z Starting task: Run PowerShell on ...
 
2015-11-23T14:43:01.5377922Z Executing the powershell script: C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\default\tasks\PowerShellOnTargetMachines\1.0.19\PowerShellOnTargetMachines.ps1
 
2015-11-23T14:43:02.0864097Z ##[debug]Entering script PowerShellOnTargetMachines.ps1
 
2015-11-23T14:43:02.0864097Z ##[debug]environmentName = ...
 
2015-11-23T14:43:02.0864097Z ##[debug]adminUserName = ...
 
2015-11-23T14:43:02.0864097Z ##[debug]protocol = Https
 
2015-11-23T14:43:02.0864097Z ##[debug]testCertificate = false
 
2015-11-23T14:43:02.0864097Z ##[debug]resourceFilteringMethod = machineNames
 
2015-11-23T14:43:02.0864097Z ##[debug]machineNames = 
 
2015-11-23T14:43:02.0864097Z ##[debug]scriptPath = c:\deploy\test.ps1
 
2015-11-23T14:43:02.0864097Z ##[debug]scriptArguments = ...
 
2015-11-23T14:43:02.0864097Z ##[debug]initializationScriptPath = 
 
2015-11-23T14:43:02.0864097Z ##[debug]runPowershellInParallel = true
 
2015-11-23T14:43:02.0864097Z ##[debug]sessionVariables = 
 
2015-11-23T14:43:02.9861307Z ##[debug]Starting Register-Environment cmdlet call for environment : ... with filter 
 
2015-11-23T14:43:03.2464788Z ##[error]A parameter cannot be found that matches parameter name 'EnvironmentName'.

**Comment by russelldyas** [comment]
I had inital problem with powershell as well but as I hadn't got that part working correctly before weekend. I decided to reenable file copy that I had stopped for debugging purposes which I knew worked..

**Comment by RoopeshNair** [comment]
I presume you're using Hosted Agents which is due for an upgrade. Post upgrade these issues should go away. Do let us know if you're facing these issue with latest on-premises agent.

Roopesh

**Comment by russelldyas** [comment]
Ok do we know when the next update is due or is there a work around for those using VSO etc. As major part of VSO release manager to be broken.

Russ

**Comment by RoopeshNair** [comment]
Will get back on update.

As a work around, you can setup on prem agent for now

**Comment by blsher** [comment]
I'm experience simliar problems, machines now doesn't accept [MachineGroup],[fqdn] the machine group specified a username and password now it wants one per step, not really happy about this.  As i don't control the passwords and don't want to have to setup a password for each step... not to mention the admin/password in the step  isn't secured, it's in clear text.

**Comment by kmkumaran** [comment]
WindowsMachineFileCopy continues to support both MachineGroup and FQDN. You can continue to use machine group and define username and password with it. As Roopesh called out the failure would be addressed with hosted agent update. You can unblock with a latest on prem agent. With regard to password not being secure, You can define secure variables in Build and refer to $(passwordVariable) as input to the password field to address password mask requirement.

**Comment by blsher** [comment]
If i give it a machinegroup name, and a "filter", [machinegroup],[machinename] so i don't have to enter a password in every step.  It asks for a password/user each step.

And i have the latest on prem agent.

**Comment by RoopeshNair** [comment]
Please specify one machine group at a time. If you've specified credentials in machine group then you need not specify it again

Multiple values are supported only for machines not for "machine groups"

**Comment by RoopeshNair** [comment]
Note you can either specify a machine group or list of machines not both.

**Comment by blsher** [comment]
{
            "name":"ResourceFilteringMethod",
            "type":"radio",
            "label":"Select Machines By",
            "required":false,
            "defaultValue": "machineNames",
            "groupName": "advanced",
            "visibleRule": "CopyFilesInParallel = Invalid",
            "options": {
                "machineNames":"Machine Names",
                "tags":"Tags"
            }
        },

That coming soon?  I really need to apply filters.

**Comment by russelldyas** [comment]
Quick question is using on-prem client with VSO what URL do you need to put in setup for the agent to connect etc.  Tried several different combinations etc.

**Comment by blsher** [comment]
I only had to use https://[myurl].visualstudio.com/ the last i remember, it's been a while since i set up a brand new agent.

**Comment by russelldyas** [comment]
If I use http://XXX.visualstudio.com:1000 vor https://XXX.visualstudio.com:1000 it will get as far as trying to config but then errors out saying can't connect.

**Comment by blsher** [comment]
I didn't use port numbers when i set up, not sure if that matters anymore.

**Comment by russelldyas** [comment]
Tried with and with out port numbers.  If I try without port numbers dosn't get passed validation stage. I wonder if before configre the deployment have to authenticate in some way to VSO?

Russ

**Comment by blsher** [comment]
Yeah your run as agent, needs build access to vso.

**Comment by blsher** [comment]
@RoopeshNair 

2015-11-25T17:29:14.7338847Z ##[debug]Starting Register-Environment cmdlet call for environment : QA_WEB01 with filter 

Can't specify filters, how do i do that now?  There's no box, nothing and the script wants it.

**Comment by blsher** [comment]
2015-11-25T17:29:14.7338847Z ##[debug]Starting Register-Environment cmdlet call for environment : QA_WEB01 with filter 

And after your update.

i can't change, modify or add filters.

**Comment by RoopeshNair** [comment]
@bisher

Did you remove the step and added it again? The filter values are in the task json, unless you removed the task and added it from the library again. 
If you're still having the original task

Please try following:
1. Ensure you've the latest version of agent. (1.91.1)
2. In Windows File Copy Task
Machines: <Machine Group Name> // sans '>'
Admin login: blank
Password: blank 
//Admin login and password details would be fetched from Machine Group definition under "Test" hub
//Filter values would be retrieved from the task json and will be applied again. 

If you've added new task then for time being you can't specify filters. As mentioned earlier, we will enable filtering capabilities in subsequent deployment. As a workaround, you can either use in-line machine list + credentials or you need to create machine group for each filter combination and specify them as above sample.

Let us know if you're still running into issues

**Comment by russelldyas** [comment]
@bisher (or anyone else) Do you mean that logged in user to OS has to be setup within VSO.  In which case I though you could add email-based accounts.  The local Azure machine is workgroup machine so can't add that user to VSO?

Russ

**Comment by russelldyas** [comment]
@RoopeshNai any update on the timescale for the fix to vso platform.  As currently having trouble getting on-prem agent to work with vso..

**Comment by blsher** [comment]
@RoopeshNair  so what your'e saying is, you removed a feature that i'm using.  And will re-release it later.  And that i can't modify any existing tasks because i can't modify the filters.

**Comment by blsher** [comment]
@russelldyas yeah was referring to on-prem agents.

**Comment by RoopeshNair** [comment]
@bisher, we're fixing the filtering issue.Shall update soon.

**Comment by RoopeshNair** [comment]
@bisher,  We've fixed it, you should be able to see the filter options in the advanced section.

You should also be able to see your current filter values. Let us know if you still have issues

**Comment by blsher** [comment]
I can see the current values now, we've been making adjustments to things and everything still seems operational.  WE did how ever notice an outage for about 3-4 hours were all our builds stopped working and the next business day they were all functional again. Didn't quite look into that one well enough.
